[2.9.0] Adding docs for agent-tls-mode (#1378)
rancher/rancher-docs: docs/getting-started/installation-and-upgrade/install-upgrade-on-a-kubernetes-cluster/troubleshooting.md

### Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?

The node is not reachable on the configured `address` and `port`.

### Agent reports TLS errors 

> When using Rancher, you may encounter errors in the fleet-agent, system-agent, or cluster-agent like the below:

Reviewer: ```suggestion
When using Rancher, you may encounter error messages from the `fleet-agent`, `system-agent`, or `cluster-agent`, such as the message below:
```

```
tls: failed to verify certificate: x509: failed to load system roots and no roots provided; readdirent /dev/null: not a directory
```

> This occurs when Rancher was configured with `agent-tls-mode` set to `strict`, but no cacerts were found in the `cacert` setting. To resolve the issue, set the `agent-tls-mode` to `system-store`, or upload the CA for rancher as described in [Adding TLS Secrets](../resources/add-tls-secrets.md).

Reviewer 2: ```suggestion
This occurs when Rancher was configured with `agent-tls-mode` set to `strict`, but no cacerts were found in the `cacert` setting. To resolve the issue, set `agent-tls-mode` to `system-store`, or upload the CA for Rancher as described in [Adding TLS Secrets](../resources/add-tls-secrets.md).
```

> This occurs when Rancher was configured with `agent-tls-mode` set to `strict`, but no cacerts were found in the `cacert` setting. To resolve the issue, set the `agent-tls-mode` to `system-store`, or upload the CA for rancher as described in [Adding TLS Secrets](../resources/add-tls-secrets.md).

Reviewer 2: ```suggestion
This occurs when Rancher was configured with `agent-tls-mode` set to `strict`, but couldn't find cacerts in the `cacert` setting. To resolve the issue, set the `agent-tls-mode` to `system-store` or upload the CA for Rancher as described in [Adding TLS Secrets](../resources/add-tls-secrets.md).
```

### New Cluster Deployment is stuck in "Waiting for Agent to check in"

> When using `agent-tls-mode` set to `strict`, new clusters may fail to provision with a generic "Waiting for Agent to check in" error message. The root casuse of this is similar o the above case for TLS errors - Rancher's agent can't determine the which CA Rancher is using (or can't verify that Rancher's cert is actually signed by the provided certificate authority).  

Reviewer 2: ```suggestion
When Rancher has `agent-tls-mode` set to `strict`, new clusters may fail to provision with a generic "Waiting for Agent to check in" error message. The root cause of this is similar to the above case of TLS errors - Rancher's agent can't determine which CA Rancher is using (or can't verify that Rancher's cert is actually signed by the provided certificate authority).  
```

> When Rancher has `agent-tls-mode` set to `strict`, new clusters may fail to provision with a generic "Waiting for Agent to check in" error message. The root cause of this is similar to the above case of TLS errors - Rancher's agent can't determine the which CA Rancher is using (or can't verify that Rancher's cert is actually signed by the provided certificate authority).  

Reviewer 2: ```suggestion
When Rancher has `agent-tls-mode` set to `strict`, new clusters may fail to provision and report a generic "Waiting for Agent to check in" error message. The root cause of this is similar to the above case of TLS errors - Rancher's agent can't determine which CA Rancher is using (or can't verify that Rancher's cert is actually signed by the specified certificate authority).  
```

> When Rancher has `agent-tls-mode` set to `strict`, new clusters may fail to provision with a generic "Waiting for Agent to check in" error message. The root cause of this is similar to the above case of TLS errors - Rancher's agent can't determine the which CA Rancher is using (or can't verify that Rancher's cert is actually signed by the provided certificate authority).  

Reviewer 3: ```suggestion
When Rancher has `agent-tls-mode` set to `strict`, new clusters may fail to provision with a generic "Waiting for Agent to check in" error message. The root cause of this is similar to the above case of TLS errors - Rancher's agent can't determine which CA Rancher is using (or can't verify that Rancher's cert is actually signed by the provided certificate authority).  
```

> To resolve the issue, set the `agent-tls-mode` to `system-store`, or upload the CA for Rancher as described in [Adding TLS Secrets](../resources/add-tls-secrets.md).

Reviewer 2: ```suggestion
To resolve the issue, set the `agent-tls-mode` to `system-store` or upload the CA for Rancher as described in [Adding TLS Secrets](../resources/add-tls-secrets.md).
```
